src/proximity-manager/index.ts

```diff
@@ -181,71 +194,83 @@ const ProximityManager = () => {
   };
 
   const handleDiscoverPeripheral = async (peripheral: Peripheral) => {
-    if (peripheral.id === randomVerifierUUID) {
-      await BleManager.stopScan();
-      await BleManager.connect(peripheral.id);
-      console.debug(`[connectPeripheral][${peripheral.id}] connected.`);
-      eventManager.emit('onEvent', {
-        type: 'ON_PERIHPERAL_CONNECTED',
-        message: 'pheripheral connected.',
-      });
-      const peripheralData = await BleManager.retrieveServices(peripheral.id);
-      console.debug(
-        `[connectPeripheral][${peripheral.id}] retrieved peripheral services`,
-        peripheralData
-      );
+    const serviceUUIDs = peripheral.advertising.serviceUUIDs;
+    if (serviceUUIDs && serviceUUIDs.includes(randomVerifierUUID)) {
+      if (!connectedPheripheral) {
+        BleManager.stopScan();
+        connectToPeripheral(peripheral);
+      }
+    }
+  };
 
-      const rssi = await BleManager.readRSSI(peripheral.id);
-      console.debug(
-        `[connectPeripheral][${peripheral.id}] retrieved current RSSI value: ${rssi}.`
-      );
+  const connectToPeripheral = async (peripheral: Peripheral) => {
+    connectedPheripheral = peripheral;
+    await BleManager.connect(peripheral.id);
+    console.debug(`[connectPeripheral][${peripheral.id}] connected.`);
+    eventManager.emit('onEvent', {
+      type: 'ON_PERIHPERAL_CONNECTED',
```

Comment:
typo... I think it also necessary to update event manager type.

```suggestion
      type: 'ON_PERIPHERAL_CONNECTED',
```